assets/js/components/ErrorNotice.js

```diff
@@ -98,26 +98,36 @@ export default function ErrorNotice( {
 	const reconnectURL = error?.data?.reconnectURL;
 
 	if ( reconnectURL && isURL( reconnectURL ) ) {
-		errorMessageWithModifications = createInterpolateElement(
+		errorMessageWithModifications =
+			errorMessageWithModifications +
+			' ' +
 			sprintf(
-				/* translators: %1$s: Original error message */
+				/* translators: %s: Reconnect URL */
 				__(
-					'%1$s To fix this, <a>redo the plugin setup</a>.',
+					'To fix this, <a href="%s">redo the plugin setup</a>.',
 					'google-site-kit'
 				),
-				errorMessageWithModifications
-			),
-			{
-				a: <Link href={ reconnectURL } />,
-			}
-		);
+				reconnectURL
+			);
 	}
 
+	const sanitizeArgs = {
+		ALLOWED_TAGS: [ 'a' ],
+		ALLOWED_ATTR: [ 'href' ],
+	};
+
 	return (
 		<Notice
 			className={ className }
 			type={ Notice.TYPES.ERROR }
-			description={ errorMessageWithModifications }
+			description={
+				<span
+					dangerouslySetInnerHTML={ sanitizeHTML(
+						errorMessageWithModifications,
+						sanitizeArgs
+					) }
+				/>
+			}
```

Comment:
I understand why this was reverted, but there's no context in the actual component. Let's add a comment here (and above, where the `<a` is in the translation string) to explain the reason we're using `sanitizeHTML`/etc.

Otherwise we'll just make the same refactoring mistake in the future 😅